pkg/boerrors/perror.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package boerrors
+
+import (
+	"fmt"
+)
+
+// BuildOpError extends standard error to:
+//  1. Keep persistent / transient property of the error.
+//     All errors, except ETransientErrorId considered persistent.
+//  2. Have error ID to show the root cause of the error and optionally short message.
+type BuildOpError struct {
+	// id is used to determine if error is persistent and to know the root cause of the error
+	id BOErrorId
+	// typically used to log the error message along with nested errors
+	err error
+}
+
+func NewBuildOpError(id BOErrorId, err error) *BuildOpError {
+	return &BuildOpError{
+		id:  id,
+		err: err,
+	}
+}
+
+func (r BuildOpError) Error() string {
+	return r.err.Error()
+}
+
+// ShortError returns short message with error ID in case of persistent error or
+// standard error message for transient errors.
+func (r BuildOpError) ShortError() string {
+	if r.id == ETransientError {
+		return r.Error()
+	}
+	return fmt.Sprintf("%d: %s", r.id, boErrorMessages[r.id])
+}
+
+func (r BuildOpError) IsPersistent() bool {
+	return r.id != ETransientError
+}
+
+type BOErrorId int
+
+const (
+	ETransientError BOErrorId = 0
```

Comment:
That way is also possible, I was thinking about it as well... but I am still in doubt which one is better.
Probably, having `ETransientError` id is a bit better, because then we will have a complete error type that can be used without relaying on another types.
However I don't have strong opinion on this, so if you do - we may change the approach.

Reply:
Make sense. Let's keep it as current :)